osu.Game/Screens/Edit/Editor.cs

```diff
@@ -1339,6 +1339,47 @@ public bool HandleTimestamp(string timestamp, bool notifyOnError = false)
 
         public int BeatDivisor => beatDivisor.Value;
 
+        #region Main UI Dimming
+
+        private bool? mainUIDimmed;
+        private readonly List<(Drawable container, Drawable[] dimmableElements)> mainUIElements = new List<(Drawable, Drawable[])>();
+
+        /// <summary>
+        /// Register an editor UI element that has background layers which can be made semi-transparent when not focused (aka hovered).
+        /// </summary>
+        /// <param name="container">The main UI element's container to be used for checking whether hovered.</param>
+        /// <param name="backgroundElements">Any elements which should have their alpha transformed when not focused.</param>
+        internal void RegisterMainUIElement(Drawable container, IEnumerable<Drawable> backgroundElements)
```

Comment:
I dunno this feels pretty dreadful to me.

- Violates encapsulation by scraping out background elements from inside components
- By changing things around this way the dimming doesn't only take place in compose tab like it used to. Stuff will dim on something like the timing screen which looks bad because the main portion of the screen is opaque while the rest is not. See the following (admittedly egregious) example:

![osu_2024-07-11_08-55-17](https://github.com/ppy/osu/assets/20418176/d3332b85-f805-43bf-a9bc-d7604a0f7985)

Using `Contains()` is the way but I argue it should be called for the _composer_ drawable rather than have all of the pieces handle it themselves. I don't really see "centralisation of transforms" as that much of a win or even an issue. If it's a design goal then we can split a const in `Editor` or something.

Reply:
fair enough